### DISCUSSION TITLE
When to use UniTask.Lazy and when to use UniTask.Preserve()

### DISCUSSION THREAD

**Discussion by yongyili** [Cysharp/UniTask#333] [Q&A]
According to the README, if I want to await a UniTask multiple times, I should use `UniTask.Layz` or `UniTask.Preserve`.
> you can use UniTask.Lazy that supports calling multiple times. .Preserve() allows for multiple calls (internally cached results). 

But what are the difference between the two methods, and how can I confirm which one is more suitable for me?

**Comment by kamyker** [comment]
+1

**Comment by ClaytonHunsinger** [comment]
I have been wondering the same thing here! I wish the developers or somebody knowledgeable could clear these things up.
That being said, I will make an educated guess based on the readme.

> Store to the class field, you can use `UniTask.Lazy` that supports calling multiple times. `.Preserve()` allows for multiple calls (internally cached results). This is useful when there are multiple calls in a function scope.

To me this almost seems as though `.Preserve()` should be used when awaiting the same UniTask multiple times inside of the same function scope. So if the UniTask is created in one function, `.Preserve()` would let it be awaited on multiple times in that same function or in any functions being called by the first function.
Meanwhile it also seems as though `.Lazy` would be used in cases where we create a UniTask, and we want to store it in a class field where it can be awaited on multiple times by different functions.

I am not saying that is correct, but given the lack of an official response I figured I'd give it my best guess.
